src/middleware/devtools.ts

```diff
@@ -165,8 +169,8 @@ const devtoolsImpl: DevtoolsImpl =
       extractConnectionInformation(store, extensionConnector, options)
 
     let isRecording = true
-    ;(api.setState as NamedSet<S>) = (state, replace, nameOrAction) => {
-      const r = set(state, replace)
+    ;(api.setState as any) = ((state, replace, nameOrAction) => {
```

Comment:
I am not sure why this workaround was needed, since a `fn(state: Partial<S>, replace: boolean | undefined, nameOrAction) should make the function definition broad enough to be valid as implementation for both overloads